tgui/packages/tgui/interfaces/MarionetteRemote.js

```diff
@@ -0,0 +1,163 @@
+                      content="Say"
+                      selected={selected_command === "say"}
+                    />
+                    <Button
+                      onClick={() => act("set_command", { new_command: "emote" })}
+                      content="Emote"
+                      selected={selected_command === "emote"}
+                    />
+                    <Button
+                      onClick={() => act("set_command", { new_command: "step" })}
+                      content="Step"
+                      selected={selected_command === "step"}
+                    />
+                    <Button
+                      onClick={() => act("set_command", { new_command: "drop" })}
+                      content="Drop"
+                      selected={selected_command === "drop"}
+                    />
+                    <Button
+                      onClick={() => act("set_command", { new_command: "use" })}
+                      content="Use"
+                      selected={selected_command === "use"}
+                    />
+                    <Button
+                      onClick={() => act("set_command", { new_command: "shock" })}
+                      content="Shock"
+                      selected={selected_command === "shock"}
+                    />
+                  </LabeledList.Item>
+                  <LabeledList.Item key="action_heat" label="Heat Per Action">
+                    {selected_command === "shock" || selected_command === "drop" ? "HIGH"
+                      : selected_command === "step" ? "LOW" : "MEDIUM"}
+                  </LabeledList.Item>
+                </LabeledList>
+              </Flex.Item>
+            </Flex>
+          </Section>
+          <Section title="Implants" buttons={(
+            <Button
+              icon="rotate"
+              content="Refresh"
+              onClick={() => act('ping_all')}
+            />
+          )}>
+            {mapImplants(act, entered_data, selected_command, implants)}
+          </Section>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const tooltipForStatus = (status) => {
+  switch (status) {
+    case "IDLE":
+      return "This implant is not located inside a living being.";
+    case "ACTIVE":
+      return "This implant is inside a living being and ready to accept signals.";
+    case "WAITING...":
+      return "Awaiting ping response...";
+    case "DANGER":
+      return "This implant is dangerously hot. Further short-term use will likely cause it to break.";
+    case "NO RESPONSE":
+      return "This implant is not responding to pings. It could have been destroyed, or it could just be far away.";
+    case "BURNED OUT":
+      return "This implant has been rendered permanently unusable by overuse and can be removed from the tracking list.";
+    default:
+      return "Unknown.";
+  }
+};
+
+const mapImplants = (act, entered_data, selected_command, implants) => {
```

Comment:
`mapImplants` should be a function Component, letting you do `<MapImplants />` above. The context is passed so you can get your variables the same way with `useBackend`.

Actually I would suggest making an `<Implant implant={implant} />` component and having the `<LabeledList>`, the `map(...)` and the no implants check be in the parent component.